### ISSUE TITLE
Compiling warning in drucker_prager material model

### ISSUE THREAD

**Issue by Shangxin-Liu** [geodynamics/aspect#1742]
There is always a warning about this material model when I compile ASPECT from git hub:

drucker_prager.cc:73:54: warning: reference cannot be bound to dereferenced
      null pointer in well-defined C++ code; comparison may be assumed to always evaluate to false [-Wtautological-undefined-compare]
              const double edot_ii_strict = (&(this->get_simulator()) == NULL
                                              ~~~~~~~^~~~~~~~~~~~~~~~    ~~~~
drucker_prager.cc:313:36: note: in instantiation of member function
      'aspect::MaterialModel::DruckerPrager<2>::evaluate' requested here
    ASPECT_REGISTER_MATERIAL_MODEL(DruckerPrager,
                                   ^
aspect/include/aspect/simulator_access.h:162:7: note: 'get_simulator' returns a reference
      get_simulator () const;
      ^

**Comment by bangerth** [comment]
The correct solution is to add a member function to `SimulatorAccess`:
```
  /**
   * Return whether or not the current object has been initialized by providing it with
   * a pointer to a Simulator class object.
   */
  bool SimulatorAccess::simulator_is_initialized() const
  {
    return (simulator != NULL);
  }
```
and then in the Drucker Prager model replace the condition by
```
  this->simulator_is_initialized()
```

Want to give this a try? That should not be much more than a 5 minute project.

**Comment by Shangxin-Liu** [comment]
I need to put the comment in the simulator_access.h header file, yes? @bangerth

**Comment by bangerth** [comment]
Put the declaration of the function (with the documentation comment) into the header file, and the implementation of the function into the source file.

**Comment by Shangxin-Liu** [comment]
OK. A pull request was already created.

**Comment by gassmoeller** [comment]
Closed by #1747
